llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2465,6 +2465,33 @@ static void licm(VPlan &Plan) {
       R.moveBefore(*Preheader, Preheader->end());
     }
   }
+
+  // Sink any recipes which don't have any users in the region to the nearest
+  // common dominator of its users.
+  VPDominatorTree VPDT(Plan);
+  for (VPBasicBlock *VPBB : VPBlockUtils::blocksOnly<VPBasicBlock>(
+           vp_post_order_shallow(LoopRegion->getEntry()))) {
+    for (VPRecipeBase &R : make_early_inc_range(reverse(*VPBB))) {
```

Comment:
Is this a reverse-post-order-shallow walk? Perhaps worth using the RPOT iterator?

Reply:
We want just regular post order here, RPOT would traverse defs before uses. The block itself is reversed though so we go from bottom to top